pysamstats/io.py

```diff
@@ -149,6 +149,10 @@ def write_hdf5(stats_type, outfile, alignmentfile, fields=None, progress=None, h
     # determine dtype
     if dtype is None:
         dtype = dict(default_dtype)
+        max_seqid_len = determine_max_seqid(alignmentfile)
+        if max_seqid_len > np.dtype(dtype["chrom"]).itemsize:
```

Comment:
I wonder if we could do away with this conditional check. I.e., just set the dtype to the max_seqid_len, ignore the default. There's no point in using e.g. 'a10' just because it's the default if the longest seqid is less than 10.